### DISCUSSION TITLE
Adding pre_ping=True causes this error: sqlalchemy.exc.ProgrammingError: `(psycopg2.ProgrammingError) set_session cannot be used inside a transaction`

### DISCUSSION THREAD

**Discussion by jtay-pm** [sqlalchemy/sqlalchemy#10986] [Usage Questions]
Adding `pre_ping=True` causes this error: `sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) set_session cannot be used inside a transaction` to trigger all the time whenever i use a commit() or connection.execute() within a transaction.

I tried looking for previous github issues on this error and saw this (https://github.com/sqlalchemy/sqlalchemy/discussions/8009) but we are not using concurrent threads so i don't think the solution for setting the thread count to 1 applies to us here..

![image](https://github.com/sqlalchemy/sqlalchemy/assets/77286898/999a0b1e-6c4d-4659-8cc3-fb98d075eaf4)

Originally posted by @jtay-pm in https://github.com/sqlalchemy/sqlalchemy/discussions/10640#discussioncomment-8323499

Using sqlalchemy 1.4.46
psycopg2 2.9.9

I'm using pre_ping to handle disconnects because I was previously getting another error: `SSL connection closed unexpectedly` when listening to connection begin events and trying to execute some queries with that connection

**Comment by zzzeek** [comment]
as mentioned in #8009 this is not an issue we can reproduce here so you'd want to look at custom on_connect handlers and such that may be getting in the way
